### ISSUE TITLE
SIGSEGV during nfs_link

### ISSUE THREAD

**Issue by ayugu** [sahlberg/libnfs#186]
Version: 1.11.0

Stack trace is below. The SEGV happened in memcmp, which tried to compare nfsdir->fh.data.data_val to fh->data.data_val, and fh->data.data_val was 0.

With a quick look at the code, it seems the bug is due to fh being stolen away in nfs_link_continue_2_internal and never assigned back.

```
    @     0x2adab0f51576 __memcmp_sse4_1
    @     0x2adaa166ea30 nfs_dircache_find.isra.15
    @     0x2adaa166efb0 nfs_dircache_drop
    @     0x2adaa1670037 nfs_link_cb
    @     0x2adaa16782fd rpc_process_pdu
    @     0x2adaa1678f92 rpc_service
    @     0x2adaa16756d2 wait_for_nfs_reply.isra.1
    @     0x2adaa16767e3 nfs_link
```

**Comment by sahlberg** [comment]
Thanks for the bug report.
I think I have a fix for this in 64f7e73ded5896b02e9a9b64471f4bc9a5ec7f4d

**Comment by ayugu** [comment]
Cool. Thanks!
